python/tests/test_nessie_cli.py

```diff
@@ -315,7 +316,36 @@ def test_merge() -> None:
     # Passing detached commit-id plus a _different_ hash-on-ref --> error
     execute_cli_command(["merge", f"dev@{dev_hash}", "-c", main_hash, "-o", main_hash], ret_val=1)
 
-    execute_cli_command(["merge", "dev", "-c", main_hash])
+    assert_that(execute_cli_command(["merge", "dev", "-c", main_hash])).starts_with("The following 1 commits were merged onto main:")
+    logs = simplejson.loads(execute_cli_command(["--json", "log"]))
+    # we don't check for equality of hashes here because a merge
+    # produces a different commit hash on the target branch
+    assert len(logs) == 1
+    logs = simplejson.loads(execute_cli_command(["--json", "log", "dev"]))
+    assert len(logs) == 1
```

Comment:
wondering: if we ran the same merge command afterwards, what kind of error would we be getting now?
maybe this could be part of the test WDYT ?

Reply:
Added the same